source/coll_db.f90

```diff
@@ -0,0 +1,598 @@
+
+  famID = cdb_getFamilyID(famName)
+  if(famID == -1) then
+    cdb_nfam = cdb_nFam + 1
+    call cdb_allocFam
+    cdb_famName(cdb_nFam) = famName
+    cdb_famNSig(cdb_nFam) = nSig
+    famID = cdb_nFam
+    fErr = .false.
+  else
+    write(lout,"(a)") "COLLDB> Warning Collimator family '"//trim(famName)//"' already exists."
+    fErr = .true.
+  end if
+
+end subroutine cdb_addFamily
+
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-03-21
+!  Updated: 2019-03-22
+!  Find a family in the database and returns its ID
+! ================================================================================================ !
+integer function cdb_getFamilyID(famName) result(famID)
+
+  character(len=*), intent(in) :: famName
+  integer i
+
+  famID = -1
+  if(cdb_nFam > 0) then
+    do i=1,cdb_nFam
+      if(cdb_famName(i) == famName) then
+        famID = i
+        exit
+      end if
+    end do
+  end if
+
+end function cdb_getFamilyID
+
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-03-21
+!  Updated: 2019-03-22
+!  Find a family in the database and returns its nsig
+! ================================================================================================ !
+real(kind=fPrec) function cdb_getFamilyNSig(famName) result(nSig)
+
+  use crcoall
+
+  character(len=*), intent(in) :: famName
+  integer famID
+
+  famID = cdb_getFamilyID(famName)
+  if(famID > 0) then
+    nSig = cdb_famNSig(famID)
+  else
+    write(lout,"(a)") "COLLDB> Warning No nsig value found for collimator family '"//trim(famName)//"'"
+    nSig = 0.0_fPrec
+  end if
+
+end function cdb_getFamilyNSig
+
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-03-21
+!  Updated: 2019-03-21
+!  Write family database to file
+! ================================================================================================ !
+subroutine cdb_writeFam
+
+  use mod_units
+
+  integer j, famUnit
+
+  call f_requestUnit("coll_families.dat", famUnit)
+  call f_open(unit=famUnit,file="coll_families.dat",formatted=.true.,mode="w",status="replace")
+
+  write(famUnit,"(a16,1x,a13)") "# famName       ","nSig"
+  do j=1,cdb_nFam
+    write(famUnit,"(a16,1x,f13.6)") cdb_famName(j),cdb_famNSig(j)
+  end do
+
+  flush(famUnit)
+  call f_close(famUnit)
+
+end subroutine cdb_writeFam
+
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-03-19
+!  Updated: 2019-03-20
+!  Extract Family Name from Old Format DB
+! ================================================================================================ !
+subroutine cdb_generateFamName(inElem, famName)
```

Comment:
Ugh. Some comment about "for backwards compatibility" would probably be a good idea... :P

Reply:
It says "from old format DB" already.